src/nvim/os/win_defs.h

```diff
@@ -15,6 +16,15 @@
 # ifndef restrict
 #  define restrict __restrict
 # endif
+# ifndef S_ISDIR
+#  define S_ISDIR(mode) (((mode) & S_IFMT) == S_IFDIR)
+# endif
+# ifndef S_ISREG
+#  define S_ISREG(mode) (((mode) & S_IFMT) == S_IFREG)
+# endif
+# ifndef S_IXUSR
+#  define S_IXUSR S_IEXEC
+# endif
```

Comment:
Unless I am mistaken the only one that is needed is this one. The others are here:

https://github.com/neovim/neovim/blob/master/src/nvim/os/os_defs.h#L54-L71